### DISCUSSION TITLE
Question about self-hosted option

### DISCUSSION THREAD

**Discussion by nchelaru** [healthchecks/healthchecks#503] [Q&A] [answered]
Hello!

First of all, thank you for making such an amazing too!

I would like to set up a self-hosted version for work, as due to security needs, we need the set-up to be self-contained. First and foremost, does healthchecks.io have access to any of our data/code?

There needs to only be a single user (the superuser) for the self-hosted option in my case. If I set `EMAIL_USE_VERIFICATION`=False, and only sign in using superuser credentials, will this mean that the set-up is self-contained on our server?

Thank you again for answering these naive questions!

**Comment by cuu508** [comment, marked as answer]
> First and foremost, does healthchecks.io have access to any of our data/code?

No, if you set up a self-hosted instance, healthchecks.io does not have any access to it. There is no phoning-home or telemetry code in the project.

> There needs to only be a single user (the superuser) for the self-hosted option in my case. If I set EMAIL_USE_VERIFICATION=False, and only sign in using superuser credentials, will this mean that the set-up is self-contained on our server?

If you only need a single user, make sure to set `REGISTRATION_OPEN=False`. This disables the "Sign Up" feature. 

When you add an email integration to a project, normally it sends a verification link, and the integration only becomes active after the verification link is clicked. There's one exception to that: if account's owner adds their own email address as an email integration, then *we skip the verification step* to save some hassle for the user. So, for a single-user usecase, setting `EMAIL_USE_VERIFICATION=False` will probably make no difference, as the email address would get auto-verified anyway.

**Reply by cuu508** [reply]
PS. on the phoning-home part: I can only vouch for the code in this repository. If you use a third party Docker container, you should check what customizations it adds on top.

**Reply by nchelaru** [reply]
Thank you very much for your quick reply and the explanations!

As far as I know, I am not using a third party Docker container. I followed the set-up instructions in the README and things seem to be working file, this should mean that I am only using code in this repo?

Finally, the limitations of the free tier also apply to the self-hosted option?

**Reply by cuu508** [reply]
Yes, if you followed the instructions in README then 3rd party customizations is not a concern.

On a self-hosted service you can set the limits yourself. The account will have a 500-check limit by default, but you can log into [the admin interface](https://github.com/healthchecks/healthchecks#accessing-administration-panel), and pick a different limit in the Profile section:

![image](https://user-images.githubusercontent.com/661859/116909742-4e3cd180-ac4d-11eb-8533-631f0b570efd.png)

**Reply by nchelaru** [reply]
Ah, thank you so much!

**Comment by iammoizazam** [comment]
Hi ,

I'm trying to deploy a self-hosted version of health checks, I've tested the app in development mode, and it's working fine great.
Thumbs UP. But I'm encountering an error when I change debug value to False it gives me "Server Error 500".
I added my domain name to the allowed hostname list. moreover when I run manage check --deploy it shows me no error.
please guide me through the solution.
thanks in advance.

**Reply by cuu508** [reply]
When DEBUG=False, and there's an exception, Django will send error reports to the email addresses listed in [ADMINS](https://docs.djangoproject.com/en/4.0/ref/settings/#admins). For this to work, you need to have outgoing email credentials configured (see the [EMAIL_ environment variables](https://healthchecks.io/docs/self_hosted_configuration/#EMAIL_HOST)).

The email reports should have clues on what's wrong and why you're getting HTTP 500 responses.

Also, have you run `manage.py collectstatic` and `manage.py compress`?

**Reply by iammoizazam** [reply]
i have run manage.py collectstatic but not manage.py compress let me try it.

**Reply by iammoizazam** [reply]
Thanks @cuu508  Issues is resolved after running `manage.py compress`

**Comment by pmatuszy** [comment]
well - you need to use e.g. gunicorn as a webserver - this is what i do. 

https://github.com/healthchecks/healthchecks/discussions/620

P.

**Comment by iammoizazam** [comment]
i'm using apache2 as a webserver and application is working fine in debug mode I put an eye on console and apache error logs there is no any error.
